js/scripts/zone_loaders/tilemap_interpreter.js

```diff
@@ -0,0 +1,153 @@
+                        break;
+                    case 'i':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANK_SPRUCE_STAIRS_LEFT));
+                        break;
+                    case 'h':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANK_OAK_STAIRS_RIGHT), 0);
+                        break;
+                    case 'g':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANK_OAK_STAIRS_LEFT));
+                        break;
+                    case 'f':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.STONE_COBBLE_VOLCANIC));
+                        break;
+                    case 'e':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.STONE_COBBLE_DARK));
+                        break;
+                    case 'd':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.STONE_COBBLE));
+                        break;
+                    case 'c':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANKS_REDWOOD_LIGHT));
+                        break;
+                    case 'b':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANKS_REDWOOD));
+                        break;
+                    case 'a':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.LOG_SPRUCE_VIRTICAL));
+                        break;
+                    case '9':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.LOG_SPRUCE_HORIZONTAL));
+                        break;
+                    case '8':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANKS_SPRUCE));
+                        break;
+                    case '7':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.PLANKS_OAK));
+                        break;
+                    case '6':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.BARS));
+                        break;
+                    case '5':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.LAVA_ROCK));
+                        break;
+                    case '4':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.SNOWY_ICE));
+                        break;
+                    case '3':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.SNOWY_DIRT));
+                        break;
+                    case '2':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.GRASS));
+                        break;
+                    case '1':
+                        GAME.addEntity(new Block(new Vector(x, y), Block.DIRT), -1);
+                        break;
+
+                    default:
+                        break;
+                }
```

Comment:
This could be simplified. Generally, if a switch is made up of one-line blocks, I prefer a different approach like building a map as you have done elsewhere.

Reply:
Got it, thanks! I was wondering about that and am not a fan of this approach either.

Reply:
I was running into issues trying to make this change. For some reason this is only making the dirt blocks. It should be interpreting all the blocks and putting them in the midground and after I figure that out i'd sort out which go to what ground and put surface_snow down only if it is snowing. I found that if I set the conditional near the bottom to != instead of == it displayed the inverse meaning all the blocks got displayed save for the dirt blocks and it seemed like air blocks with bounding boxes were made as well. Can you see what I'm missing? For now I'll finish pushing my other changes and come back to this if time permits.
 `const blocks = {
            'S': Block.CAVE_SHARP_UP_GROUP_GREY,
            'R': Block.CAVE_SHARP_DOWN_GROUP_GREY,
            'Q': Block.CAVE_SHARP_UP_1_GREY,
            'P': Block.CAVE_SHARP_DOWN_1_GREY,
            'O': Block.CAVE_SHARP_UP_GROUP,
            'N': Block.CAVE_SHARP_DOWN_GROUP,
            'M': Block.CAVE_SHARP_UP_1,
            'L': Block.CAVE_SHARP_DOWN_1,
            'K': Block.CAVE_4,
            'J': Block.CAVE_3,
            'I': Block.CAVE_2,
            'H': Block.CAVE_1,
            'G': Block.STONE_COBBLE_DARK,
            'F': Block.BR_RIGHT,
            'E': Block.BR_LEFT,
            'D': Block.BR_RIGHT_END,
            'C': Block.BR_LEFT_END,
            'A': Block.TWIG_LEFT,
            'B': Block.TWIG_RIGHT,
            'z': Block.LOG_SPRUCE_VIRTICAL,
            'y': Block.LOG_SPRUCE_VIRTICAL,
            'x': Decoration.DECORATIONS.grass.GRASS_1,
            '@': Decoration.DECORATIONS.grass.GRASS_1,
            'w': Decoration.DECORATIONS.grass.GRASS_3,
            'v': Block.TWIG_LEFT,
            'u': Block.TWIG_RIGHT,
            't': Block.BR_RIGHT_HALF,
            's': Block.BR_RIGHT_FULL,
            'r': Block.BR_LEFT_HALF,
            'q': Block.BR_LEFT_FULL,
            'p': Block.BRANCH_TOP,
            'o': Block.BRANCH_FILL_LOG,
            'n': Block.BRANCH_FILL,
            'm': Block.DIRT,
            'l': Block.SNOW_SURFACE,
            'k': Block.ICE,
            'j': Block.PLANK_SPRUCE_STAIRS_RIGHT,
            'i': Block.PLANK_SPRUCE_STAIRS_LEFT,
            'h': Block.PLANK_OAK_STAIRS_RIGHT,
            'g': Block.PLANK_OAK_STAIRS_LEFT,
            'f': Block.STONE_COBBLE_VOLCANIC,
            'e': Block.STONE_COBBLE_DARK,
            'd': Block.STONE_COBBLE,
            'c': Block.PLANKS_REDWOOD_LIGHT,
            'b': Block.PLANKS_REDWOOD,
            'a': Block.LOG_SPRUCE_VIRTICAL,
            '9': Block.LOG_SPRUCE_HORIZONTAL,
            '8': Block.PLANKS_SPRUCE,
            '7': Block.PLANKS_OAK,
            '6': Block.BARS,
            '5': Block.LAVA_ROCK,
            '4': Block.SNOWY_ICE,
            '3': Block.SNOWY_DIRT,
            '2': Block.GRASS,
            '1': Block.DIRT
        };
        for (let y = ZONE.MAX_BLOCK.y; y >= ZONE.MIN_BLOCK.y; y--) {
            for (let x = ZONE.MAX_BLOCK.x; x >= ZONE.MIN_BLOCK.x; x--) {
                // console.log(this.tilemap[y][x]);
                // console.log(blocks[this.tilemap[y][x]]);
                let curBlock = this.tilemap[y][x];
                console.log(curBlock);
                if (curBlock == blocks[curBlock]) {
                    GAME.addEntity(new Block(new Vector(x, y), blocks[curBlock]));
                }
    }
}`